### PR TITLE
Bugfix: allow root of N5 container to be a dataset when parsed with base "/"

### PR DESCRIPTION
**Pull request by cmhulbert**
If the N5 container root is a dataset, `getName()` on the root metadata throws an exception, see https://github.com/saalfeldlab/n5-imglib2/issues/33.

Additionally, this trims whitespace on N5TreeNode creation.

### CODE REVIEW

**Review by bogovicj — COMMENTED** [review]
I believe we implemented `discoverAndParseRecursive` such that its root metadata path matches the string you passed to it (i.e. "/" or ""), right?

This is reasonable, but let's think about whether that's what we want.
And let's test for whatever behavior we decide.  What do you think?

File: src/test/java/org/janelia/saalfeldlab/n5/metadata/MetadataTests.java [hunk 1]
```diff
@@ -276,4 +276,99 @@ public void testGenericMetadata() {
...
+			Arrays.asList(parsers),
+			Arrays.asList(grpparsers));
+
+	for (String base : new String[]{"/", "   /   ", "/\n", "\t\n\t/\t"}) {
+
+	  try {
+		final N5TreeNode n5root = discoverer.discoverAndParseRecursive(base);
+
+		Assert.assertNotNull(n5root.getPath(), n5root.getMetadata());
+		Assert.assertTrue("is multiscale cosem", n5root.getMetadata() instanceof N5CosemMultiScaleMetadata);
```
[the remark this answers is not in the document]
Comment (cmhulbert) [reply src/test/java/org/janelia/saalfeldlab/n5/metadata/MetadataTests.java:300]: I think this is the same as I added a bit below, on line 304. Is it preferable to have it higher?

File: src/test/java/org/janelia/saalfeldlab/n5/metadata/MetadataTests.java [hunk 2]
```diff
@@ -276,4 +276,99 @@ public void testGenericMetadata() {
...
+			Arrays.asList(parsers),
+			Arrays.asList(grpparsers));
+
+	for (String base : new String[]{"", "   ", "\n", "\t\n\t \t"}) {
+
+	  try {
+		final N5TreeNode n5root = discoverer.discoverAndParseRecursive(base);
+
+		Assert.assertNotNull(n5root.getPath(), n5root.getMetadata());
+		Assert.assertTrue("is multiscale cosem", n5root.getMetadata() instanceof N5CosemMultiScaleMetadata);
```
[the remark this answers is not in the document]
Comment (cmhulbert) [reply src/test/java/org/janelia/saalfeldlab/n5/metadata/MetadataTests.java:347]: Similar to above, I have this on like 351.

**Comment by cmhulbert** [comment]
I think for now that returning the basePath for the root metadata is what we want in the sense that it's consistent with all other basePath. I'm not sure if it's what we want from a more holistic perspective. I think for my usecases, I'm happy with it, but If you have other suggestions, I'd be interested.

**Comment by bogovicj** [comment]
Looks good, let's merge this

**Pull request merged**